src/drivers/console.c

```diff
@@ -16,18 +16,56 @@ struct console_driver serial_console_driver = {
     .name = "serial"
 };
 
-struct console_driver *console_drivers[4] = {
+struct console_driver *console_drivers[8] = {
     &serial_console_driver,
 };
 
 static struct spinlock write_lock;
 
+static void sort_console_drivers(void)
```

Comment:
This is not critical, but I think it would be simpler to keep the console_drivers on a list, with list heads for both enabled and disabled. Or use a single list and always move disabled drivers to the end (and move to head on enable). Then use list_foreach in console_write and break once a disabled entry is found.

Reply:
OK, I changed the console code to use a single list, with disabled drivers at the end.